src/features/game/dialogue/GameDialogueManager.ts

```diff
@@ -102,8 +105,43 @@ export default class DialogueManager {
     }
   }
 
+  /**
+   * Hide all dialogue boxes, speaker boxes and speaker sprites
+   * */
+  public async hideAll() {
+    await this.getDialogueRenderer().hide();
+    await this.getSpeakerRenderer().hide();
+  }
+
+  /**
+   * Make all dialogue boxes, speaker boxes and speaker sprites visible
+   * */
+  public async showAll() {
+    await this.getDialogueRenderer().show();
+    await this.getSpeakerRenderer().show();
+  }
+
+  /**
+   * Find patterns of quiz score interpolation in a dialogue line,
+   * and replace them by actual scores.
+   * The pattern: "{<quizId>.score}"
+   *
+   * @param line
+   * @returns {string} the given line with all quiz score interpolation replaced by actual scores.
+   */
+  public makeLineWithQuizScores(line: string) {
+    const quizScores = line.match(/\{.+?\.score\}/g);
+    if (quizScores) {
+      quizScores.forEach(match => {
+        const quizId = match.substring(1, match.lastIndexOf('.'));
+        line = line.replace(match, GameGlobalAPI.getInstance().getQuizScore(quizId).toString());
+      });
+    }
+    return line;
```

Comment:
Seems like using `matchAll` and a regex capturing group can simplify the logic by a lot here, thoughts @lhw-1?

Reply:
```suggestion
function makeLineWithQuizScores(line : string) {
    const quizScores = line.matchAll(/\{.+?\.score\}/g);
    for (const match of quizScores) {
        line = line.replace(match[0], 
            GameGlobalAPI.getInstance()
            .getQuizScore(match[0].substring(1, match[0].lastIndexOf('.')))
            .toString())
    }
}

```
is this a valid improvement?

Reply:
```suggestion
    const quizScores = line.matchAll(/\{(.+?)\.score\}/g);
    for (const match of quizScores) {
      line = line.replace(match[0], GameGlobalAPI.getInstance().getQuizScore(match[1]).toString());
    }
    return line;
```
I guess this will work? Used capturing group to remove the use of `substring`.

Reply:
The version suggested by @reginateh seems to be the most concise 👍